### PR TITLE
Added ability to profile nested lists

### PR DESCRIPTION
**Pull request by zslade**
Addresses issue [#1034](https://github.com/moj-analytical-services/splink/issues/1034).

Can now supply nested lists to `linker.profile_columns()` to profile multiple columns alongside singular columns. This allows users to profile the multiple columns that make up a blocking rule.

This does not currently work for `sqlite`

### CODE REVIEW

**Review by ADBond — CHANGES_REQUESTED** [review]
I think something may be going amiss with the `NULL` handling here. Not sure exactly what, but if I save one of the new charts from `test_profile_data.py` (with a couple of additions) I get this:
profile
which seems to suggest that in these cases the `NULL` values are being treated as blank strings, although they are handled correctly if you pass individual columns (not as a singleton array).

File: splink/profile_data.py [hunk 1]
```diff
@@ -129,11 +142,44 @@ def _get_df_top_bottom_n(expressions, limit=20, value_order="desc"):
...
+            null_exprs = " OR ".join(null_exprs)
+
+            col_or_expr = f"""
+                case when
+                {null_exprs} then null
+                else
+                {col_or_expr}
+                end
+            """
+
```
[the remark this answers is not in the document]
Comment (ThomasHepworth) [reply splink/profile_data.py:182]: Yup. The rebase didn't go to plan😳

**Comment by ThomasHepworth** [comment]
Ta Andy. It's odd that the `NULL` handling has now decided to stop working.

You can see the old results in the issue I originally posted.

I'll take a look and see if I can figure it out.

**Comment by ThomasHepworth** [comment]
@ADBond found the [issue](https://github.com/moj-analytical-services/splink/pull/1074/commits/3412205cfb2155d4a30fe9dbf130a7aec27dfa3b).

It was a slight error when the rebase happened.

**Review by ADBond — APPROVED** [review]
Looks good, working fine now 👍 . Seems there are just a couple of small issues with the imports in test file causing the linter check to fail, but happy for this to be merged once those are fixed.

**Pull request merged**